### PR TITLE
Adds a connection for fav'd genes

### PR DESCRIPTION
**Pull request by orta**
Adds a connection for genes:

```graphql
{
  me {
    followed_genes(first: 5) {
      edges {
        node {
          id
          gene {
            name
          }
        }
      }
    }
  }
}

```

Equals:
```
{
  "data": {
    "me": {
      "followed_genes": {
        "edges": [
          {
            "node": {
              "id": "5a0df980275b245c42586afe",
              "gene": {
                "name": "Engaged with Traditional African Art"
              }
            }
          },
          {
            "node": {
              "id": "5a0df9828b3b817683082bb9",
              "gene": {
                "name": "Animals"
              }
            }
          },
          {
            "node": {
              "id": "5a0df9838b3b81766a9785d0",
              "gene": {
                "name": "Group of Portraits"
              }
            }
          }
        ]
      }
    }
  }
}
```

Also updates the artist followed to be a dataloader.

### CODE REVIEW

File: schema/me/followed_genes.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import _ from "lodash"
```
Comment (mzikherman) [inline schema/me/followed_genes.js]: can just be `import { has } from...`

File: lib/loaders/loaders_with_authentication/gravity.js [hunk 2]
```diff
@@ -31,6 +31,8 @@ export default (accessToken, userID, requestID) => {
     ),
     saleArtworksLoader: gravityLoader(id => `sale/${id}/sale_artworks`),
     suggestedArtistsLoader: gravityLoader("me/suggested/artists", {}, { headers: true }),
+    myFollowedArtistsLoader: gravityLoader("me/follow/artists", {}, { headers: true }),
+    myFollowedGenesLoader: gravityLoader("me/follow/genes", {}, { headers: true }),
```
Comment (alloy) [inline lib/loaders/loaders_with_authentication/gravity.js]: Why do you feel this needs a `my` prefix? The user specific loaders will always be about the authenticated user anyways.

File: schema/gene.js [hunk 3]
```diff
@@ -28,7 +28,7 @@ const SUBJECT_MATTER_REGEX = new RegExp(SUBJECT_MATTER_MATCHES.join("|"), "i")
 const GeneType = new GraphQLObjectType({
   name: "Gene",
   interfaces: [NodeInterface],
-  isTypeOf: obj => _.has(obj, "family") && _.has(obj, "browseable"),
+  isTypeOf: obj => _.has(obj, "browseable") && (_.has(obj, "published") || _.has(obj, "family")),
```
Comment (alloy) [inline schema/gene.js]: Why is this change necessary?

**Comment by alloy** [comment]
None of the comments are blockers imo, feel free to merge whenever 👍

File: schema/gene.js (hunk 3 above)
Comment (orta) [reply schema/gene.js]: The short version of the gene here does not [include the family](https://github.com/artsy/metaphysics/blob/genes_cinnection/test/fixtures/gravity/follow_genes.json#L1) but I think in the other place it's used in the home page, it always includes family

Comment (alloy) [reply schema/gene.js]: Aha, gotcha. I hope the `browseable` and `published` aren’t too generic that they could potentially exist in other payloads, though.

In either case, I don’t think it’s really necessary to do an `or`, just choose a better alternative and use that for all cases.
